reverse-proxy-nginx.conf

```diff
@@ -0,0 +1,70 @@
+user nginx;
+worker_processes auto;
+error_log /var/log/nginx/error.log warn;
+pid /var/run/nginx.pid;
+events {
+  worker_connections  1024;
+}
+http {
+  resolver 127.0.0.11;
+  include /etc/nginx/mime.types;
+  default_type application/octet-stream;
+  log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                    '$status $body_bytes_sent "$http_referer" '
+                    '"$http_user_agent" "$http_x_forwarded_for"';
+  access_log /var/log/nginx/access.log  main;
+  sendfile on;
+  keepalive_timeout 65;
+  server {
+    listen 3000;
+    server_name localhost;
+
+    location / {
+      proxy_pass http://frontend:3000;
+    }
+    
+    location /api/ {
+      # the Set-Cookie: XDEBUG_SESSION=PHPSTORM; path=/; SameSite=Lax header is set too many times
+      # temporary workaround from https://stackoverflow.com/a/27551259
+      proxy_buffer_size 128k;
+      proxy_buffers 4 256k;
+      proxy_busy_buffers_size 256k;
+      proxy_set_header X-Forwarded-Prefix /api;
+      proxy_pass http://api:3001/;
+    }
+    
+    location /print {
+      proxy_pass http://print:3003;
+    }
+    
+    location /mail {
+      proxy_pass http://mail:1080;
+    }
+  }
+  server {
+    listen 3004;
+    server_name localhost;
+
+    location / {
+      proxy_pass http://frontend:3000;
+    }
+    
+    location /api/ {
+      # the Set-Cookie: XDEBUG_SESSION=PHPSTORM; path=/; SameSite=Lax header is set too many times
+      # temporary workaround from https://stackoverflow.com/a/27551259
+      proxy_buffer_size 128k;
+      proxy_buffers 4 256k;
+      proxy_busy_buffers_size 256k;
+      proxy_set_header X-Forwarded-Prefix /api;
+      proxy_pass http://http-cache:8080/;
```

Comment:
Is this workaround still needed?

Reply:
Yes, we get too many headers from the upstream for the default buffer.